README.md

```diff
@@ -1,5 +1,278 @@
+> "I am inevitable!"
+>           - A mean purple titan
+
 # Overview
 
-Library for constructing structured text.
+Constructor library for structured monospaced text.
+
 Current functionality:
-* Pending...
+* Multi-line text of uniform character width, and arbitrary padding
+* Multi-line columnated-row text with vertical dividers
+* Textual table with horizontal dividers, header, and footer
+* Blocked text with border, title, and caption
+
+Each of these elements can be combined to create structured text that can be printed to a console, or used to display arbitrarily sized text data to the screen.
+
+*Note*: The display font must render with a monospaced font for the text to align as intended.
+
+## Kotlin & Java Compatibility
+
+This library was constructed using Kotlin, and as such used several idiomatic Kotlin structures for the API. However, it was also designed to emulate a standard Java builder pattern structure for the API. So each construct can be called using either approach.
+
+### Target Compatibility
+
+|        | Version |
+|--------|---------|
+| JVM    | `1.9`   |
+| Kotlin | `1.3.x` |
+
+## Usage Examples
+
+*Note*: These examples can be composed with each other in arbitrary ways to render the structured text.
+
+### Multi-line Uniform Text
+
+#### Kotlin API
+
+Let the pre-constructed line values be:
+```kotlin
+val line1 = faker.getWords(5)
+val line2 = faker.getWords(2)
+val line3 = faker.getWords(3)
+```
+
+Then, these values can be used in the following ways:
+
+##### Option 1
+```kotlin
+textCell(line1, line2, line3) {
+    withAlignment(CENTER)
+    withPadding(1, 1)
+    withPaddingSpacer('*')
+}.println()
+```
+
+##### Option 2
+```kotlin
+textCell(listOf(line1, line2, line3)) {
+    withAlignment(CENTER)
+    withPadding(1, 1)
+    withPaddingSpacer('*')
+}.println()
+```
+
+##### Option 3
+```kotlin
+textCell {
+    withLines(line1, line2, line3)
+    withAlignment(CENTER)
+    withPadding(1, 1)
+    withPaddingSpacer('*')
+}.println()
+```
+
+##### Option 4
+```kotlin
+textCell {
+    addLines(line1)
+    addLines(line2)
+    addLines(line3)
+    withAlignment(CENTER)
+    withPadding(1, 1)
+    withPaddingSpacer('*')
+}.println()
+```
+
+#### Java API
+
```

Comment:
missing java block?

Reply:
I haven't quite filled this out yet. For some reason I can't get the Java tests to run on concert with the kotlin tests. It breaks all the tests whenever I add a Java test. Not sure why

Reply:
Do we need two separate executions? ones that use `src/test/kotlin/**` and ones that use `src/test/java/**` ?

Reply:
I wanted to create some Java tests so I can call the code using the Java style using the API tests. Since the Kotlin syntax is a bit different, I think it helps to see it both ways.